### ISSUE TITLE
How to save scroll position while navigating between pages using Link from gatsby?

### ISSUE THREAD

**Issue by SilencerWeb** [gatsbyjs/gatsby#26654]
##  Summary
As I remember, before, the scroll position was always saved while navigating between pages using `Link` from `gatsby`. I remember there were a lot of compliments about it and questions how to prevent this behavior since mostly Gatsby was used for creating websites, not web apps and such behavior is not really common and useful for websites. And right it seems like Gatsby does not save scroll position anymore? But if need this behavior for some pages? Currently, I need it with the usage of `navigate` from `gatsby` and wondering if it is possible to achieve this behavior?

**Comment by ezvr** [comment]
Has there been any further explanation for this? I'm also using gatsby to build a website, which requires that the scroll position is retained.

**Comment by benjmo** [comment]
I'm actually trying to do the opposite - make a traditional website experience where scroll is not retained.
But I get the scroll retaining behaviour when `overflow: auto` or `overflow: hidden` is set on the HTML `body`.

I was under the impression that this was fixed?

**Comment by blainekasten** [comment]
@benjmo please open a different issue if you need help with that.

@ezvr and @SilencerWeb gatsby should be retaining the scroll position by default with the same behavior that a browser does when javascript is not involved.

One thing you can do to check is look at localStorage when you're navigating. You should see some keys added for each history element and it's scroll position. Do you not see that? What version of gatsby are you running? Maybe more specifically what version of `gatsby-react-router-scroll` do you have in your node_modules (use `yarn why` or `npm ls`)

**Comment by ezvr** [comment]
@blainekasten 
> One thing you can do to check is look at localStorage when you're navigating. You should see some keys added for each history element and it's scroll position. Do you not see that? 

Nothing is being added to my localStorage. You can observe the problematic website on https://kisik.si/tehnologija/. Try clicking on the menu items on the left, while being scrolled. 



>What version of gatsby are you running? Maybe more specifically what version of `gatsby-react-router-scroll` do you have in your node_modules (use `yarn why` or `npm ls`)

```
=> Found "gatsby-react-router-scroll@3.0.12"
info Reasons this module exists
   - "gatsby" depends on it
   - Hoisted from "gatsby#gatsby-react-router-scroll"

```

**Comment by micha149** [comment]
Are you guys scrolling on the window? When setting body to something like `height: 100%` and defining a custom scroll container, the default scroll tracking will not work. Its looking to scroll events on your window object, but they occur on your scroll container.

You can use the `useScrollRestoration` hook and attach it to your custom scroll container. But be aware: there is currently a problem when using it in a layout component, which I fixed in #26861.
